### PR TITLE
Memory usage warning

### PR DESCRIPTION
**Pull request by trapvincenzo**
This method allows you to check if the memory usage exceeded a specific threshold.

### CODE REVIEW

File: src/utilphp/util.php [hunk 1]
```diff
@@ -2540,4 +2548,31 @@ public static function directory_contents($dir)
...
+     * Checks if the memory usage exceeded a specific threshold.
+     * 
+     * @param int $warning_memory_limit The warning memory limit (in percentage)
+     * @param int $php_memory_limit Php memory limit
+     * @return bool
+     */
+    public static function memory_usage_warning($warning_memory_limit = null, $php_memory_limit = null)
+    {
+        if (is_null($warning_memory_limit)) {
+            $warning_memory_limit = self::$warning_memory_limit;
```
Comment (lyrixx) [inline src/utilphp/util.php]: What's the point to add a new static variable? why don't use a default value for this function?

Comment (trapvincenzo) [inline src/utilphp/util.php]: Hi,
using a static allows you to manage a global warning limit for your application (maybe different for each environment of your application).

With a static you can do
```php
util::warning_memory_limit = 50;
```
and be sure that everytime and everywhere you call the memory_usage_warning the limit is the same.

Comment (lyrixx) [inline src/utilphp/util.php]: but the property is private, so it's not possible (unless you are using the reflection)

File: src/utilphp/util.php [hunk 2]
```diff
@@ -2540,4 +2548,31 @@ public static function directory_contents($dir)
...
+
+        if (is_null($php_memory_limit)) {
+            $php_memory_limit = PHP_INT_MAX;
+        }
+
+        $threshold = intval(($warning_memory_limit * $php_memory_limit) / 100);
+        $used_memory = memory_get_usage(true);
+
+        //Exceeded
+        if ($used_memory >= $threshold)
```
Comment (lyrixx) [inline src/utilphp/util.php]: `return $used_memory >= $threshold` is enough.

File: src/utilphp/util.php [hunk 3]
```diff
@@ -2540,4 +2548,31 @@ public static function directory_contents($dir)
...
+        if (is_null($warning_memory_limit)) {
+            $warning_memory_limit = self::$warning_memory_limit;
+        }
+
+        if (is_null($php_memory_limit)) {
+            $php_memory_limit = PHP_INT_MAX;
+        }
+
+        $threshold = intval(($warning_memory_limit * $php_memory_limit) / 100);
+        $used_memory = memory_get_usage(true);
```
Comment (lyrixx) [inline src/utilphp/util.php]: IMHO, you should just use `memory_get_usage()`

File: src/utilphp/util.php (hunk 1 above)
Comment (trapvincenzo) [inline src/utilphp/util.php]: Sorry, fixed.

**Comment by brandonwamboldt** [comment]
Don't think this is useful enough to include

**Pull request closed without merging**